09/src/javascripts/wd.js

```diff
@@ -1,48 +1,46 @@
 module.exports = function () {
     let apps = [];
-
-    let app = {
-        appRootName: null,
-        appRoot: null,
-        appRoutes: [],
-        createRoot: function (rootName) {
-            let root = document.querySelector("main[wd-root=" + rootName + "]");
+    let context = {
```

Comment:
Instead of object literal, you should create constructor `function App()`, or `class App {...}`. 
It will have method `createRoot()` which returns instance of the constructor or class.
```javascript
// something like that
function App(){
  /* props */
}

App.prototype.createRoot = function(name) {
   const app = new App(name);
   this.apps.push(app); 
   return app;
}

```